server/src/main/java/org/apache/druid/segment/handoff/CoordinatorBasedSegmentHandoffNotifier.java

```diff
@@ -70,7 +86,7 @@ public boolean registerSegmentHandoffCallback(SegmentDescriptor descriptor, Exec
   @Override
   public void start()
   {
-    scheduledExecutor = Execs.scheduledSingleThreaded("coordinator_handoff_scheduled_%d");
+    scheduledExecutor = Execs.scheduledSingleThreaded("[" + StringUtils.encodeForFormat(indexTaskId) + "]-coordinator_handoff_scheduled_%d");
```

Comment:
Rather than changing the thread name, I would advise including the task ID in the log lines like:

```
"Still waiting for handoff of [%d] segments for task[%s]",
handoffCallbacks.size(), taskId
```

Reply:
Can I do both? It will help if handoff notifier thread goes to any exception or unusual flow

Reply:
Would including the `taskId` in the exception message (the same way that `dataSource` and `segmentId` is included) not be enough?

Reply:
Your argument is valid. I thought of adding `taskId` to process thread name, as StreamAppenderator Service with threads like `[TASK_ID]-publish`, `[TASK_ID]-appenderator-persist`.
But there are very few log lines in Coordinator Segment handoff notifier, so it would be better to add taskId in log itself

Reply:
Hey @kfaraz ! Thanks for the suggestions. I have addressed all comments